README.md

```diff
@@ -0,0 +1,178 @@
+# slsa-github-generator
+
+This repository contains a reference implementation for generating non-forgeable
+[SLSA provenance](https://slsa.dev/) that meets the requirements for the [SLSA
+level 2 and above](https://slsa.dev/spec/v0.1/levels) using a [Github Actions
+reusable
+workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows).
+
+---
+
+- [Generating Provenance](#generating-provenance)
+  - [Getting Started](#getting-started)
+  - [Workflow Inputs](#workflow-inputs)
+  - [Provenance Example](#provenance-example)
+- [Verification of Provenance](#verification-of-provenance)
+  - [Inputs](#inputs)
+  - [Command-line Examples](#command-line-examples)
+
+---
+
+## Generating Provenance
+
+`slsa-github-generator` uses a Github Actions reusable workflow to generate the
+provenance.
+
+### Getting Started
+
+To get started, you will need to add some steps to your current workflow. We
+will assume you have an existing Github Actions workflow to build your project.
+
+Add a step to your workflow after you have built your project to generate a
+sha256 hash of your artifacts. The following assumes you have a binary called
+`binary-linux-amd64`.
+
+After that, add a new job to call the `slsa-github-generator` reusable workflow.
+
+```yaml
+jobs:
+  build:
+    outputs:
+      digest: ${{ steps.hash.outputs.digest }}
+    runs-on: ubuntu-latest
+    steps:
+      # Your build steps are here.
+      - name: "generate hash"
+        shell: bash
+        id: hash
+        run: |
+          set -euo pipefail
+          DIGEST=$(sha256sum binary-linux-amd64)
+          DIGEST="${DIGEST//'%'/'%25'}"
+          DIGEST="${DIGEST//$'\n'/'%0A'}"
+          DIGEST="${DIGEST//$'\r'/'%0D'}"
+          echo "::set-output name=digest::$DIGEST"
+  provenance:
+    needs: [build]
+    permissions:
+      id-token: write
+      contents: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/provenance.yml@main
```

Comment:
Let's rename this provenance.yml to `slsa2_provenance.yml`?

Reply:
Ok, do you foresee us providing others like slsa3_provenance.yml? Just trying to understand the reasoning.